### PR TITLE
MYG-8542 zombie to Puppeteer

### PR DESCRIPTION
**Pull request by NolanKingdon**
Other pull request was based off of my forked master. This should solve the issue

### CODE REVIEW

File: generators/app/templates/test/functional/test.js [hunk 1]
```diff
@@ -15,68 +15,109 @@ const rpcResponse = (response, err) => {
...
 
+// puppeteer options
+const opts = {
+    headless: true,
+    slowMo: 0,
+    timeout: 10000
+};
+
 // Mocks
 const mockServer = mocks.server;
```
Comment (beefo) [inline generators/app/templates/test/functional/test.js]: Looks like these variables are no longer used.

File: generators/app/templates/test/functional/test.js [hunk 2]
```diff
@@ -15,68 +15,109 @@ const rpcResponse = (response, err) => {
...
+    before(async () => {
+        browser = await puppeteer.launch(opts);
+        const context = browser.defaultBrowserContext();
+        page = await browser.newPage();
+        // Allowing puppeteer access to the request - needed for mocks
+        await page.setRequestInterception(true);
+
+        // Setup mocks
+        await page.on('request', request => {
+            if (request.url() === `https://${mocks.server}/apiv1`) {
```
Comment (beefo) [inline generators/app/templates/test/functional/test.js]: Found this was never true. URL seemed to be http:// for me. Did you find the same?

File: generators/app/templates/test/functional/test.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
-const Browser = require('zombie');
-const nock = require('nock');
+const puppeteer = require('puppeteer');
 const mocks = require('./mocks/mocks.js');
+const assert = require('chai').assert;
 
 // JSON-RPC helpers
 const rpcRequest = body => {
```
Comment (beefo) [inline generators/app/templates/test/functional/test.js]: Was getting an error here about invalid json. This seemed to fix for me:
```
const rpcRequest = body => {   
    let decodedBody = decodeURIComponent(body);
    let json = decodedBody.replace('JSON-RPC=', '');
    return JSON.parse(json);
};
```

File: generators/app/templates/test/functional/test.js (hunk 1 above)
Comment (NolanKingdon) [reply generators/app/templates/test/functional/test.js]: Confirmed and removed

File: generators/app/templates/test/functional/test.js (hunk 2 above)
Comment (NolanKingdon) [reply generators/app/templates/test/functional/test.js]: Did further testing and found the same. Fixed the issue.

File: generators/app/templates/test/functional/test.js (hunk 3 above)
Comment (NolanKingdon) [reply generators/app/templates/test/functional/test.js]: Also worked on my branch

**Pull request merged**